OpenRA.Game/Actor.cs

```diff
@@ -40,34 +40,38 @@ public sealed class Actor : IScriptBindable, IScriptNotifyBind, ILuaTableBinding
 		public Group Group;
 		public int Generation;
 
-		Lazy<Rectangle> bounds;
-		Lazy<Rectangle> visualBounds;
-		Lazy<IFacing> facing;
-		Lazy<Health> health;
-		Lazy<IOccupySpace> occupySpace;
-		Lazy<IEffectiveOwner> effectiveOwner;
-
-		public Rectangle Bounds { get { return bounds.Value; } }
-		public Rectangle VisualBounds { get { return visualBounds.Value; } }
-		public IOccupySpace OccupiesSpace { get { return occupySpace.Value; } }
-		public IEffectiveOwner EffectiveOwner { get { return effectiveOwner.Value; } }
+		public Rectangle Bounds { get; private set; }
+		public Rectangle VisualBounds { get; private set; }
+		public IEffectiveOwner EffectiveOwner { get; private set; }
+		public IOccupySpace OccupiesSpace
+		{
+			get
+			{
+				if (occupySpace == null)
+					occupySpace = Trait<IOccupySpace>();
+				return occupySpace;
+			}
+		}
 
 		public bool IsIdle { get { return currentActivity == null; } }
-		public bool IsDead { get { return Disposed || (health.Value == null ? false : health.Value.IsDead); } }
+		public bool IsDead { get { return Disposed || (health != null && health.IsDead); } }
 
-		public CPos Location { get { return occupySpace.Value.TopLeft; } }
-		public WPos CenterPosition { get { return occupySpace.Value.CenterPosition; } }
+		public CPos Location { get { return OccupiesSpace.TopLeft; } }
+		public WPos CenterPosition { get { return OccupiesSpace.CenterPosition; } }
 
 		public WRot Orientation
 		{
 			get
 			{
 				// TODO: Support non-zero pitch/roll in IFacing (IOrientation?)
-				var facingValue = facing.Value != null ? facing.Value.Facing : 0;
+				var facingValue = facing != null ? facing.Facing : 0;
 				return new WRot(WAngle.Zero, WAngle.Zero, WAngle.FromFacing(facingValue));
 			}
 		}
 
+		IOccupySpace occupySpace;
```

Comment:
Several traits access properties that depend on ```IOccupySpace``` during construction. So it needs to be available at that time.